### PR TITLE
LPS-79490 Add unit tests

### PR DESCRIPTION
**Pull request by rafaprax**

### CODE REVIEW

**Comment by rafaprax** [comment]
ci:test:relevant

**Comment by pedroqueiroz** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java [hunk 1]
```diff
@@ -111,6 +118,24 @@ public DDMDataProviderResponse getData(
...
+		DDMDataProviderRequest ddmDataProviderRequest,
+		DDMRESTDataProviderSettings ddmRESTDataProviderSettings) {
+
+		Map<String, String> pathParameters = getPathParameters(
+			ddmDataProviderRequest, ddmRESTDataProviderSettings);
+
+		String url = ddmRESTDataProviderSettings.url();
+
+		for (Entry<String, String> pathParameter : pathParameters.entrySet()) {
+			url = StringUtil.replaceFirst(
```
Comment (pedroqueiroz) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java]: hey @rafaprax , should we replace only the first occurrence?

Comment (rafaprax) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java]: @pedroqueiroz I think so. It is weird having a mapping like `api/{param1}/something/{param1}`. What do you think?

Comment (pedroqueiroz) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java]: It makes sense! :)

**Comment by pedroqueiroz** [comment]
Sent to BChan at https://github.com/brianchandotcom/liferay-portal/pull/58199.

**Pull request closed without merging**